polynote-kernel/src/main/scala/polynote/kernel/task/package.scala

```diff
@@ -143,12 +156,11 @@ package object task {
               .either
               .tap(_.fold(err => fail(Cause.fail(err)), _ => complete)) <* updater.join
           }
-          myTurn        <- Promise.make[Throwable, Unit]
-          imDone        <- Promise.make[Throwable, Unit]
           _             <- readyQueue.offer(QueuedTask(id, myTurn, imDone))
-          runTask        = (myTurn.await.onInterrupt(imDone.interrupt.unit) *> statusRef.update(_.running) *> taskBody)
-            .ensuring(imDone.succeed(()))
+          wait           = myTurn.await.onInterrupt(imDone.interrupt *> onUnqueue)
```

Comment:
I had to add this `onUnqueue` effect, because for some reason, interruption of `await` doesn't result in interruption of the overall inner task (so we can't handle this just with `onInterrupt` in `queueCell`)

Reply:
Never mind – I realized why the `onInterrupt` things weren't being executed, and it's because the inner task itself wasn't actually being executed.

Someone told me at one point that `queue` was confusing, maybe this was what they were talking about 🤔